sdk/offchain/src/jws.rs

```diff
@@ -0,0 +1,179 @@
+use ed25519_dalek::{ExpandedSecretKey, PublicKey, SecretKey, Signature};
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use std::{convert::TryFrom, str};
+
+#[derive(Deserialize, Serialize)]
+enum JwsAlgorithm {
+    EdDSA,
+}
+
+#[derive(Deserialize, Serialize)]
+struct Header {
+    alg: JwsAlgorithm,
+}
+
+pub fn serialize<T: Serialize>(t: &T, secret_key: &SecretKey) -> Result<String, JwsError> {
+    let mut msg = String::new();
+    base64::encode_config_buf(
+        serde_json::to_string(&Header {
+            alg: JwsAlgorithm::EdDSA,
+        })
+        .map_err(JwsError::json)?,
+        base64::URL_SAFE,
+        &mut msg,
+    );
+    msg.push('.');
+    base64::encode_config_buf(
+        serde_json::to_string(t).map_err(JwsError::json)?,
+        base64::URL_SAFE,
+        &mut msg,
+    );
+
+    // Sign msg
+    let expanded_secret_key = ExpandedSecretKey::from(secret_key);
+    let public_key = PublicKey::from(secret_key);
+    let signature = expanded_secret_key.sign(msg.as_bytes(), &public_key);
```

Comment:
might make sense to expand this into an API that supports signing and one that supports getting the signable data, to be signed outside this code, and then brought back to include the signature.

Reply:
completely agree, we probably need to do a better job at using the [`Signer`](https://docs.rs/signature/1.3.0/signature/trait.Signer.html) trait for this